### ISSUE TITLE
Dictionaries get resolved to arrays of key type by default.

### ISSUE THREAD

**Issue by Tobriand** [z4kn4fein/stashbox#67]
There's some decidedly odd behaviour in place when resolving dictionaries. I've a feeling (though I'm not sure) that this is because `UnknownTypeResolver` gets hit *after* the `EnumerableResolver` _and_ the `EnumerableResolver` simply tests whether the type to be resolved is an `IEnumerable` via 

```
public bool CanUseForResolution(IContainerContext containerContext, TypeInformation typeInfo, ResolutionContext resolutionContext) =>
    typeInfo.Type.GetEnumerableType() != null;
```

I would expect (though can't say it's right) that there would also be a test for something like "Am I trying to resolve a concrete type or not?". As it happens, though, it decides the best resolution is a `string[]` instead of a `Dictionary<string, object>` in the below.

```
            // Works
            StashboxContainer sb2 = new StashboxContainer();
            sb2.Register<IDictionary<string, object>>(typeof(Dictionary<string, object>), 
                // Without this, the default selection rule is for services which are undefined. 
                c => c.WithConstructorSelectionRule(Rules.ConstructorSelection.PreferLeastParameters)
                );
            IDictionary<string, object> dict2 = (IDictionary<string, object>)sb2.Resolve(typeof(IDictionary<string, object>));



            // Fails
            StashboxContainer sb = new StashboxContainer(
                c => c.WithUnknownTypeResolution(
                    // Should do the same as the above? However, this line is never hit.
                    c2 => c2.WithConstructorSelectionRule(Rules.ConstructorSelection.PreferLeastParameters)
                    )
                );
            Dictionary<string, object> dict = sb.Resolve<Dictionary<string, object>>();
```

**Comment by z4kn4fein** [comment]
Hey,

The fix for this one is available in the latest [pre-release package](https://www.myget.org/feed/pcsajtai/package/nuget/Stashbox), could you please check that it works as expected? Thanks!

**Comment by Tobriand** [comment]
This looks like it's working properly now on the prerelease version

**Comment by z4kn4fein** [comment]
released in [v2.8.6](https://github.com/z4kn4fein/stashbox/releases/tag/2.8.6)
